app/src/main/java/com/kabouzeid/gramophone/dialogs/DeleteSongsDialog.java

```diff
@@ -54,14 +67,80 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
                 .content(content)
                 .positiveText(R.string.delete_action)
                 .negativeText(android.R.string.cancel)
+                .autoDismiss(false)
                 .onPositive(new MaterialDialog.SingleButtonCallback() {
                     @Override
                     public void onClick(@NonNull MaterialDialog dialog, @NonNull DialogAction which) {
-                        if (getActivity() == null)
+                        Activity activity = getActivity();
+
+                        if (activity == null)
                             return;
-                        MusicUtil.deleteTracks(getActivity(), songs);
+
+                        songsToRemove = songs;
+
+                        if (!SAFUtil.isSAFRequiredForSongs(songs)) {
+                            deleteSongs(songs, null);
+                            dismiss();
```

Comment:
That or a loading animation such as the one used in the tag editor "saving changes" dialog.